src/Attributes/ParameterTrait.php

```diff
@@ -40,7 +40,7 @@ public function __construct(
     ) {
         parent::__construct([
                 'parameter' => $parameter ?? Generator::UNDEFINED,
-                'name' => $name ?? Generator::UNDEFINED,
+                'name' => Generator::isDefault($this->name) ? $name : $this->name,
```

Comment:
Ability to change the name.
Example:
```php
public function update(
        #[CookieParameter] string $cookie,
        // with custom name
        #[HeaderParameter(name: 'test-header')] ?string $header,
    ) {}
```

Reply:
I thought I mentioned that adding all sorts of shortcut versions is a slippery slope. It's a lot of classes for a small amount of syntactic sugar.

If we add them, then  yes, there will need to be annotation versions as well.

Refactoring to merge annotations and attributes is not really possible without major BC issues.
For one thing constructors would need to match and that just doesn't add up.

Reply:
I'm just making these changes in a separate branch, to see how it behaves. In general, it turns out, but the main problem is that you need to switch to a new version of the Annotation format. [Here is the documentation](https://www.doctrine-project.org/projects/doctrine-annotations/en/1.13/custom.html#optional-constructors-with-named-parameters). So, need to change a lot of examples :D in the repo.

Reply:
When working with method parameters, this is extremely necessary, because you won't be able to name the variables the same as the name of the cookie